File: Controllers/TrainerController.cpp
#include "TrainerController.h"
#include <iostream>
using namespace std;

TrainerController::TrainerController(Database& db) : db(db) {}

void TrainerController::getAll() {
    cout << "\n--- ALL TRAINERS ---\n";
    db.loadTrainers();
}

void TrainerController::getOne(int trainerID) {
    cout << "\n--- TRAINER DETAILS ---\n";
    db.getTrainerByID(trainerID);
}

void TrainerController::create(int id, string name, string email, string specialty) {
    Trainer t(id, name, specialty);
    db.saveTrainer(t, email);
    cout << "✔ Trainer '" << name << "' added successfully.\n";
}

void TrainerController::update(int trainerID, string name, string email, string specialty) {
    db.updateTrainer(trainerID, name, email, specialty);
    cout << "✔ Trainer ID " << trainerID << " updated successfully.\n";
}

void TrainerController::remove(int trainerID) {
    db.deleteTrainer(trainerID);
    cout << "✔ Trainer ID " << trainerID << " deleted. Assigned members set to no trainer.\n";
}

File: Controllers/TrainerController.js
// TrainerController.js
// Replace `trainers` array with backend API calls later

let trainers = [];

const TrainerController = {

    getAll() {
        return trainers;
    },

    getOne(trainerID) {
        return trainers.find(t => t.trainerID === trainerID) || null;
    },

    create(trainerID, name, email, specialty) {
        const exists = trainers.some(t => t.trainerID === trainerID);
        if (exists) {
            alert("Trainer ID already exists!");
            return false;
        }
        trainers.push({ trainerID, name, email, specialty });
        return true;
    },

    update(trainerID, name, email, specialty) {
        const index = trainers.findIndex(t => t.trainerID === trainerID);
        if (index === -1) {
            alert("Trainer not found!");
            return false;
        }
        trainers[index] = { trainerID, name, email, specialty };
        return true;
    },

    remove(trainerID) {
        const index = trainers.findIndex(t => t.trainerID === trainerID);
        if (index === -1) {
            alert("Trainer not found!");
            return false;
        }
        // Set assigned members' trainerID to null (mirrors ON DELETE SET NULL)
        members.forEach(m => {
            if (m.trainerID === trainerID) m.trainerID = null;
        });
        trainers.splice(index, 1);
        return true;
    },

    search(name) {
        return trainers.filter(t => t.name.toLowerCase() === name.toLowerCase());
    }
};